src/sql/platform/serialization/common/serializationService.ts

```diff
@@ -8,25 +8,56 @@ import { createDecorator } from 'vs/platform/instantiation/common/instantiation'
 import { IConnectionManagementService } from 'sql/platform/connection/common/connectionManagement';
 import { ICapabilitiesService } from 'sql/platform/capabilities/common/capabilitiesService';
 import * as azdata from 'azdata';
+import { localize } from 'vs/nls';
+import { getErrorMessage } from 'vs/base/common/errors';
 
 export const SERVICE_ID = 'serializationService';
 
-export interface SerializationProviderEvents {
-	onSaveAs(saveFormat: string, savePath: string, results: string, appendToFile: boolean): Thenable<azdata.SaveResultRequestResult>;
-}
-
 export const ISerializationService = createDecorator<ISerializationService>(SERVICE_ID);
+const saveAsNotSupported = localize('saveAsNotSupported', "Saving results into different format disabled for this data provider.");
+const defaultBatchSize = 500;
+
+export interface SerializeDataParams {
+	/**
+	 * 'csv', 'json', 'excel', 'xml'
+	 */
+	saveFormat: string;
+	filePath: string;
+	/**
+	 * Gets an array of rows to be sent for serialization
+	 * @param rowStart Index in the array to start copying rows from
+	 * @param numberOfRows Total number of rows to copy. If 0 or undefined, will copy all
+	 */
+	getRowRange(rowStart: number, numberOfRows?: number): azdata.DbCellValue[][];
+	rowCount: number;
+	columns: azdata.IDbColumn[];
+	includeHeaders?: boolean;
+	delimiter?: string;
+	lineSeperator?: string;
+	textIdentifier?: string;
+	encoding?: string;
+	formatted?: boolean;
+}
 
 export interface ISerializationService {
 	_serviceBrand: any;
 
+	registerProvider(providerId: string, provider: azdata.SerializationProvider): void;
+
+	hasProvider(): boolean;
+
 	saveAs(saveFormat: string, savePath: string, results: string, appendToFile: boolean): Thenable<azdata.SaveResultRequestResult>;
 
 	disabledSaveAs(): Thenable<azdata.SaveResultRequestResult>;
 
-	addEventListener(handle: number, events: SerializationProviderEvents): IDisposable;
+	serializeResults(request: SerializeDataParams): Promise<azdata.SerializeDataResult>;
+
+	getSaveResultsFeatureMetadataProvider(ownerUri: string): azdata.FeatureMetadataProvider;
+}
 
-	getSerializationFeatureMetadataProvider(ownerUri: string): azdata.FeatureMetadataProvider;
+function getRowCount(totalRows: number, currentIndex: number): number {
```

Comment:
maybe `getBatchSize` or `getBatchRowCount` is more descriptive name.

Reply:
getBatchSize sounds good.